src/tests/system/tests/test_sss_override.py

```diff
@@ -0,0 +1,274 @@
+        6. The name and overriden name is found and POSIX attributes are updated
+    :customerscenario: False
+    :requirement: True
+    """
+    provider.user("user1").add(
+        uid=999011, gid=999011, home="/home/user1", gecos="user", shell="/bin/bash", password="Secret123"
+    )
+
+    client.sssd.domain["ldap_id_mapping"] = "False"
+    client.sssd.config_apply()
+    client.sssd.start()
+
+    user = client.sss_override.user("user1")
+
+    user.add(name="o-user1")
+    client.sssd.restart()
+
+    assert client.auth.ssh.password("user1", "Secret123")
+    assert client.auth.ssh.password(f"user1@{client.sssd.default_domain}", "Secret123")
+    assert client.auth.ssh.password("o-user1", "Secret123")
+    assert client.auth.ssh.password(f"o-user1@{client.sssd.default_domain}", "Secret123")
+
+    result = client.tools.getent.passwd("o-user1")
+    assert result.uid == 999011
+    assert result.gid == 999011
+    user.add(name="o-user1", uid=999999, gid=888888, home="/home/o-user1")
+
+    result = client.tools.getent.passwd("user1")
+    assert result.uid == 999999
+    assert result.gid == 888888
+    assert result.home == "/home/o-user1"
+
+    result = client.tools.getent.passwd("o-user1")
+    assert result.uid == 999999
+    assert result.gid == 888888
+    assert result.home == "/home/o-user1"
+
+    user.delete()
+
+
+@pytest.mark.topology(KnownTopology.LDAP)
+@pytest.mark.topology(KnownTopology.AD)
+@pytest.mark.topology(KnownTopology.Samba)
+def test_sss_overrides__overriding_group_name_and_gid(client: Client, provider: GenericProvider):
+    """
+    :title: Locally overriding the name and GID of a group
+    :setup:
+        1. Create group "group1" with posix attributes defined
+        2. Start SSSD
+    :steps:
+        1. Create local override "group1" and name it "o-group1"
+        2: Restart SSSD, this is necessary to enable local overrides
+        3. Query groups by the local override name
+        4. Override the GID for the group "group1"
+        5. Query groups by the override name
+    :expectedresults:
+        1. Group local override is created
+        2. SSSD has restarted successfully
+        3. Group is found by the overriden name "o-group1"
+        4. Local override POSIX attribute updated
+        5. Group is found by the overriden name "o-group1" and GID changed
+    :customerscenario: False
+    :requirement: True
+    """
+    provider.group("group1").add(gid=999999)
+    client.sssd.domain["ldap_id_mapping"] = "False"
+    client.sssd.config_apply()
+    client.sssd.start()
+
+    group = client.sss_override.group("group1")
+
+    group.add(name="o-group1")
+    client.sssd.restart()
+
+    result = client.tools.getent.group("group1")
+    assert result is not None
+    assert result.gid == 999999
+    assert client.tools.getent.group("o-group1")
+    group.add(name="o-group1", gid=888888)
+
+    assert client.tools.getent.group("group1")
+    assert client.tools.getent.group("o-group1")
```

Comment:
The user test is different. It's tested twice because the override is added and then modified. We can assume that the uid/gid is what we set it when creating the user.